lib/src/model/puzzle/puzzle_providers.dart

```diff
@@ -34,11 +35,16 @@ Future<PuzzleStreakResponse> streak(StreakRef ref) {
   return Result.release(repo.streak());
 }
 
-// TODO when history database is available should first try to fetch from there
 @Riverpod(keepAlive: true)
-Future<Puzzle> puzzle(PuzzleRef ref, PuzzleId id) {
-  final repo = ref.watch(puzzleRepositoryProvider);
-  return Result.release(repo.fetch(id));
+Future<Puzzle?> puzzle(
+  PuzzleRef ref,
+  PuzzleId id,
+) {
+  // ignore: avoid_manual_providers_as_generated_provider_dependency
+  final repo = ref.watch(puzzleHistoryStorageProvider);
+  return repo.fetchPuzzle(
```

Comment:
this provider can be used to get puzzles never played before on the app so it should first try to fetch locally in the db, then if not found fetch from lichess server